### PR TITLE
mdr+cnv test case io and data integrity checks on VMs

### PR DESCRIPTION
**Pull request by parikshithb**
- [x] Need to update vm-workloads on ocs-workloads master branch
- [x] Doc text improvements

### CODE REVIEW

File: ocs_ci/helpers/cnv_helpers.py [hunk 1]
```diff
@@ -164,19 +164,26 @@ def convert_ssh_key_to_base64(ssh_key):
...
 
     """
     secret_data = templating.load_yaml(constants.CNV_VM_SECRET_YAML)
-    secret_data["metadata"]["name"] = create_unique_resource_name("vm-test", "secret")
+    if secret_name:
+        secret_data["metadata"]["name"] = secret_name
+    else:
+        secret_data["metadata"]["name"] = create_unique_resource_name(
+            "vm-test", "secret"
+        )
```
Comment (PrasadDesala) [inline ocs_ci/helpers/cnv_helpers.py:186]:
```suggestion
secret_name = secret_name if secret_name else create_unique_resource_name("vm-test", "secret")
secret_data["metadata"]["name"] = secret_name
```

File: ocs_ci/helpers/cnv_helpers.py [hunk 2]
```diff
@@ -328,3 +335,53 @@ def get_ssh_private_key_path():
...
+        username (str, optional): The username to use for SSH authentication. Defaults to None.
+        verify (bool, optional): Whether to verify the I/O operation by calculating MD5 checksum.
+            Defaults to False.
+
+    Returns:
+        str or None: If verify is True, returns the MD5 checksum of the written file. Otherwise, None.
+
+    """
+    vm_obj.run_ssh_cmd(
+        command=f"dd if=/dev/zero of=/{file_name} bs={bs} count={count}",
```
Comment (PrasadDesala) [inline ocs_ci/helpers/cnv_helpers.py:380]: Please avoid hardcoding the mount point, as the mount point of the VM disk may not always be the root

File: ocs_ci/helpers/cnv_helpers.py [hunk 3]
```diff
@@ -328,3 +335,53 @@ def get_ssh_private_key_path():
...
+    """
+    md5sum_out = vm_obj.run_ssh_cmd(
+        command=f"md5sum {file_name}",
+        username=username,
+    )
+    return md5sum_out.split()[0]
+
+
+def run_dd_io(
+    vm_obj, file_name, bs="1024", count="102400", username=None, verify=False
```
Comment (PrasadDesala) [inline ocs_ci/helpers/cnv_helpers.py:361]: Do we really need both bs and count arguments? Shouldn't we simplify it by just asking the user for the file size this function has to create?

File: ocs_ci/helpers/cnv_helpers.py (hunk 2 above)
Comment (parikshithb) [reply ocs_ci/helpers/cnv_helpers.py]: ack, added filepath

File: ocs_ci/helpers/cnv_helpers.py (hunk 3 above)
Comment (parikshithb) [reply ocs_ci/helpers/cnv_helpers.py]: Made bs as default to 1MB and count(aka size) is size_in_megabytes.

File: ocs_ci/helpers/cnv_helpers.py (hunk 1 above)
Comment (parikshithb) [reply ocs_ci/helpers/cnv_helpers.py]: ack, made the change

**Comment by parikshithb** [comment]
Verified here: https://url.corp.redhat.com/5604927

File: tests/functional/disaster-recovery/metro-dr/test_cnv_app_failover_relocate.py [hunk 4]
```diff
@@ -159,14 +187,12 @@ def test_cnv_app_failover_relocate(
...
-        # TODO: Validate Data integrity
-
         # Un-fence the managed cluster
         enable_unfence(drcluster_name=self.primary_cluster_name)
 
         # Reboot the nodes after unfenced
         gracefully_reboot_ocp_nodes(
-            self.wl_namespace, self.primary_cluster_name, cnv_workloads[0].workload_type
+            drcluster_name=self.primary_cluster_name,
         )
```
[the remark this answers is not in the document]
Comment (parikshithb) [reply tests/functional/disaster-recovery/metro-dr/test_cnv_app_failover_relocate.py:208]: ack, added.

**Comment by parikshithb** [comment]
Verified again on MDR 4.15.0-158 and cnv 4.15.1-171 https://url.corp.redhat.com/cc70cac

**Review by Akarsha-rai — APPROVED** [review]

**Review by PrasadDesala — APPROVED** [review]

**Pull request merged**